mapproject.h

```diff
@@ -87,9 +87,22 @@ but are not directly exposed by the mapscript module
     int wellknownprojection; ///< One of ``wkp_none 0``, ``wkp_lonlat 1``, or ``wkp_gmerc 2``
   } projectionObj;
 
+  typedef struct {
```

Comment:
We could likely just use a forward declaration here "typedef struct reprojectionObj reprojectionObj;" here, and let the definition in mapproject.c, since nobody needs to inspect the content of the fields.

Reply:
That's what was there originally right? Swig choked on it.

Reply:
hum, there's maybe a subtelty. In GDAL, we used forward declarations and SWIG, but I see that the type mapped to SWIG is actually a pointer to a struct, not the struct itself.
```
/** Opaque type for C++ GDALMDArray */
typedef struct GDALMDArrayHS* GDALMDArrayH;
```
(The issue of exposing the structure content is that it is not the same depending on PROJ >= 6 or PROJ < 6.)

So maybe have in mapproject.h
typedef struct reprojectorObj* reprojectObjHandle;

and in swiginc/reprojection.i

```
class reprojectorObj {
private:
  reprojectorObj();
public:
%extend {
....
}
```

Reply:
Maybe we need to give up on proj < 6 altogether?

Reply:
> we need to give up on proj < 6 altogether?

with that commit, they will certainly break due 2 incompatible definitions of struct reprojectionObj (the PROJ 6 one in mapproject.h and the remaining PROJ4 one in mapproject.c)

Reply:
I fiddled with the suggestion:

`typedef struct reprojectorObj* reprojectObjHandle;`

but ran into (what look to be) complicated Swig issues. 

I hadn't noticed the second reprojectionObj definition in mapproject.c. Could both the Proj 6 and Proj 4 definitions both be moved to mapproject.h?

Reply:
> Could both the Proj 6 and Proj 4 definitions both be moved to mapproject.h?

yes, that's doable.
But I'm wondering if the plan couldn't be to have in mapproject.h just
```
#ifdef SWIG
typedef struct {
   %immutable;
   int opaque; // to have a non empty structure
} reprojectionObj;
#endif
```

that would only be used by SWIG,and keep the actual definitions in mapproject.c
But that's still unsatisfactory. GDAL SWIG binding clearly shows there's a way to have opaque structures mapped to SWIG, although I know fighting SWIG is not fun :-)

Reply:
The error I received with the opaque structure mapping (and definitions in mapproject.c) was "Warning 303: %extend defined for an undeclared class reprojectionObj.". Could the form of the structure declarations in mapproject.c have anything to do with it:

```
struct reprojectionObj
{
    projectionObj* in;
    projectionObj* out;
    PJ* pj;
    int should_do_line_cutting;
    shapeObj splitShape;
    int bFreePJ;
};
```
versus:
```
typedef struct
{
    projectionObj* in;
    projectionObj* out;
    PJ* pj;
    int should_do_line_cutting;
    shapeObj splitShape;
    int bFreePJ;
} reprojectionObj;
```

Reply:
@rouault, does the latest change look ok to you? Looks like @jmckenna's PHP tests ran fine.